proof_system/tests/blind_bbs_plus_sig_and_message_equality.rs

```diff
@@ -0,0 +1,155 @@
+// cargo-fmt makes this much harder to read
+use ark_bls12_381::{Bls12_381, Fr};
+use ark_std::{
+    collections::{BTreeMap, BTreeSet},
+    rand::{prelude::StdRng, SeedableRng},
+    UniformRand,
+};
+use bbs_plus::prelude::SignatureG1;
+use blake2::Blake2b512;
+use proof_system::{
+    prelude::{EqualWitnesses, MetaStatements, Proof, ProofSystemError, WitnessRef, Witnesses},
+    proof_spec::ProofSpec,
+    statement::{
+        bbs_plus::{
+            PoKBBSSignatureG1Prover as PoKSignatureBBSG1ProverStmt,
+            PoKBBSSignatureG1Verifier as PoKSignatureBBSG1VerifierStmt,
+        },
+        Statements,
+    },
+    witness::PoKBBSSignatureG1 as PoKSignatureBBSG1Wit,
+};
+use test_utils::bbs::*;
+
+#[test]
+fn blind_bbs_plus_sig_and_message_equality() {
+    // This test
+    // - creates two issuers/signers
+    // - creates one signature for each signer, including a common attribute, blinded (by the holder) for each signature
+    // - it then unblinds the signatures and
+    // - creates a proof of knowledge of both signatures and equality of the blinded attributes
+    // It tests that
+    // - when the blinded values are   equal, proof creation and verification both succeeds
+    // - when the blinded values are unequal, either proof creation fails, or verification of the created proof fails
+    //
+    // This test was motivated by the fact that the existing blinding tests
+    //   proof_system/tests/bbs_plus_and_accumulators.rs
+    //     requesting_partially_blind_bbs_plus_sig
+    //     requesting_partially_blind_bbs_plus_sig
+    //     requesting_partially_blind_bbs_sig
+    // do not use the unblinded/blind signature in proofs.  Hence, this test.
```

Comment:
I didn't use the "unblinded" sig in proof (of knowledge of sig) as other tests do that and an "unblinded" sig is same as a regular sig

Reply:
The `requesting_partially_blind_bbs_sig` test doesn't unblind the blinded signature.

The `requesting_partially_blind_bbs_plus_sig` test does, and then _verifies_ the unblinded signature, but does not create any proofs with it.  One might argue that verifying the unblinded signature is sufficient to ensure it's "just the same as a regular sig", and therefore can be used to create all the same proofs.  However, I think that it's worthwhile explicitly testing that we can use the unblinded signature to create and verify at least one proof.

Reply:
`requesting_partially_blind_bbs_sig ` doesn't have the blinding term (`s`) and thus no `unblind` function for BBS sigs. The user might separately add blinding to achieve the same effect as BBS+